### DISCUSSION TITLE
pages/thoughts

### DISCUSSION THREAD

**Discussion by giscus** [whexy/whexy-blog-comments#5] [General]
# pages/thoughts/

a student obsessed with the computing world

http://localhost:8080/pages/thoughts/

**Comment by whexy** [comment]
在开发者的一片骂声中，Swift终于把部分runtime下放到旧版系统了。语言的运行时系统和OS一起shipping是很迷惑的事情。更迷惑的事情是即使履行“向下兼容”，运行时仍然依赖特定的系统版本。无论如何，async下放到iOS13相当于将工业界产品向前推动了两年。我们用不着感谢Swift团队，Tensorflow for Swift和AWS Swift已经证明：如果开源社区再这样被苹果牵着鼻子走，只会迎来没落。

**Comment by whexy** [comment]
iPad装机必备软件notability改订阅制了？从手写笔记角度，notability确实是做得最棒的，订阅是他们的底气。毕竟苹果官网宣传iPad的时候也得用他们软件的截图。购买转订阅的阵痛已经发生无数次，尤其要面对老用户的不满。作为有应用上架的开发者一员，我理解订阅制是激励维护和功能更新的最好手段。notability完全可以出一个订阅制的2.0。原版保持买断，不再更新维护。现在这么做的，一个原因当然是恰钱。另一个悲哀的原因就是，notability的开发团队已经没有办法想出创新功能，激励用户从1.0付费升级了。iPad笔记应用的赛道可能已经饱和。下一步就是各类安卓平板了。

**Reply by whexy** [reply]
Notability 在一片骂声之中终于妥协，见[推特地址](https://twitter.com/NotabilityApp/status/1455683644945678337)。
>We are making changes to our subscription so that our existing users will have lifetime access to their current features and content.

**Comment by whexy** [comment]
[Tiny PNG](https://tinypng.com/) is just AWESOME! It helps me to reduce the size of a WEBP image that I used as my homepage background from 1.3M to 26KB. That's a 10-point improvement in PageSpeed Insight!

**Comment by whexy** [comment]
Wrote an image generator. This image is dynamically generated and would be auto refreshed everytime when you visit this thread. 
![](https://whexy.com/intro.svg?)
